pkg/controller/service.go

```diff
@@ -64,8 +64,8 @@ func (*Service) Name() string {
 func (s *Service) Run(ctx context.Context) error {
 	logger := ucplog.FromContextOrDiscard(ctx)
 
-	metricsAddr := "0" // Disable metrics
-	if s.Options.Config.MetricsProvider.Prometheus.Enabled {
+	metricsAddr := "0" // Setting the address to 0 will disable the metrics server.
+	if s.Options.Config.MetricsProvider.Enabled {
```

Comment:
Note, the k8s controllers expose metrics differently because it's provided by controller-runtime.